From 377cff2e7dab6497ff21fb18923baea305778720 Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Thu, 25 Jun 2015 16:28:20 -0400
Subject: [PATCH 300/343] amd/dal: Add cursor hot-spot programming

Cursor hot-spot programming is needed when display is rotated.

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Reviewed-by: Jordan Lazare <jordan.lazare@amd.com>
Signed-off-by: Sanjay R Mehta <sanju.mehta@amd.com>
---
 drivers/gpu/drm/amd/dal/controller/cursor.c   | 3 +++
 drivers/gpu/drm/amd/dal/include/plane_types.h | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/controller/cursor.c b/drivers/gpu/drm/amd/dal/controller/cursor.c
index 458e9e5..a14eeec 100644
--- a/drivers/gpu/drm/amd/dal/controller/cursor.c
+++ b/drivers/gpu/drm/amd/dal/controller/cursor.c
@@ -55,6 +55,9 @@ bool dal_cursor_set_position(
 
 	cur->funcs->program_position(cur, position->x, position->y);
 
+	if (position->hot_spot_enable)
+		cur->funcs->program_hotspot(cur, position->x_origin, position->y_origin);
+
 	/* unlock cursor registers */
 	cur->funcs->lock(cur, false);
 
diff --git a/drivers/gpu/drm/amd/dal/include/plane_types.h b/drivers/gpu/drm/amd/dal/include/plane_types.h
index eb8201d..66b8e00 100644
--- a/drivers/gpu/drm/amd/dal/include/plane_types.h
+++ b/drivers/gpu/drm/amd/dal/include/plane_types.h
@@ -684,8 +684,14 @@ struct cursor_position {
 	uint32_t x;
 	uint32_t y;
 
+	uint32_t x_origin;
+	uint32_t y_origin;
+
 	/*This parameter indicates whether HW cursor should be enabled */
 	bool enable;
+
+	/* This parameter indicates whether cursor hot spot should be programmed */
+	bool hot_spot_enable;
 };
 
 /* This enum is for programming CURSOR_MODE register field. */
-- 
1.9.1

